sycl/include/sycl/detail/array.hpp

```diff
@@ -9,7 +9,10 @@
 #pragma once
 
 #include <sycl/detail/defines_elementary.hpp> // for __SYCL_ALWAYS_INLINE
+
+#ifndef __SYCL_DEVICE_ONLY__
```

Comment:
Not necessary because the use is guarded already:

https://github.com/intel/llvm/blob/6272defd70340ffe4952fc344b9be27c465452cf/sycl/include/sycl/detail/array.hpp#L107-L116